SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialVirtualWorkerTranslator.cpp

```diff
@@ -260,13 +271,22 @@ void SpatialVirtualWorkerTranslator::WorkerEntityQueryDelegate(const Worker_Enti
 void SpatialVirtualWorkerTranslator::AssignWorker(const PhysicalWorkerName& Name)
 {
 	check(!UnassignedVirtualWorkers.IsEmpty());
-	check(NetDriver->StaticComponentView->HasAuthority(SpatialConstants::INITIAL_VIRTUAL_WORKER_TRANSLATOR_ENTITY_ID, SpatialConstants::VIRTUAL_WORKER_TRANSLATION_MAPPING_ID));
+	check(NetDriver->StaticComponentView->HasAuthority(SpatialConstants::INITIAL_VIRTUAL_WORKER_TRANSLATOR_ENTITY_ID, SpatialConstants::VIRTUAL_WORKER_TRANSLATION_COMPONENT_ID));
 
 	// Get a VirtualWorkerId from the list of unassigned work.
 	VirtualWorkerId Id;
 	UnassignedVirtualWorkers.Dequeue(Id);
 
 	VirtualToPhysicalWorkerMapping.Add(Id, Name);
 
+	if (Name == WorkerId)
+	{
+		LocalVirtualWorkerId = Id;
+		bIsReady = true;
```

Comment:
So is this ready necessary?

Reply:
Additionally we should probably pull out a function that does:

void Function(ID, Name)
{
        VirtualToPhysicalWorkerMapping.Add(Id, Name);
	if (Name == WorkerId)
	{
		LocalVirtualWorkerId = Id;
		bIsReady = true;
		NetDriver->LoadBalanceStrategy->SetLocalVirtualWorkerId(LocalVirtualWorkerId);
	}
}

This can be called from AssignWorker and ApplyMappingFromSchema.

Reply:
Updated this.